### ISSUE TITLE
Mutliple PLEX servers

### ISSUE THREAD

**Issue by djguu** [Ombrelin/plex-rich-presence#9]
```
Authentication failed : org.simpleframework.xml.core.PersistenceException: Element 'Server' is already used with @org.simpleframework.xml.Element(name=Server, type=class fr.arsenelapostolet.plexrichpresence.model.Server, data=false, required=false) on field 'Server' private fr.arsenelapostolet.plexrichpresence.model.Server fr.arsenelapostolet.plexrichpresence.model.MediaContainerServer.Server at line 4
```

**Comment by Ombrelin** [comment]
Yes it's a know issue : #6

**Comment by Ombrelin** [comment]
Yes it's the multiple plex server error, I forgot about it

**Comment by djguu** [comment]
Thank you @Ombrelin !

**Comment by Ombrelin** [comment]
The bug has been fixed, the first plex server is now selected. I'll improve that either by polling all the user's server or by featuring a server selection UI.

**Comment by djguu** [comment]
@Ombrelin I tried to login but it doesnt log me in
`INFO  - Authentication failed : Unable to connect to plex server.`

**Comment by Ombrelin** [comment]
Mmmh I tested with multiple server myself, might be something else. We'll release a new version soon with a new login technique so maybe it'll fix this. Keep me posted if it doesn't !

**Comment by djguu** [comment]
I'll try to be on top when you release the new version :) Sometimes I skip the notifications, too many of them

**Comment by djguu** [comment]
@Ombrelin it's working now :D
Slight issue (not for me), your app uses 300mb+ of ram, I can see a problem with preople with less then 16gb of ram

**Comment by djguu** [comment]
Oh and, if I'm not on the plex app or the web interface, it says that I'm idle on discord, is that how it's supposed to be? Not really sure how hard it is to detect that you are on plex app or not

**Comment by mclarence** [comment]
> Oh and, if I'm not on the plex app or the web interface, it says that I'm idle on discord, is that how it's supposed to be? Not really sure how hard it is to detect that you are on plex app or not

I just implemented this.

@Ombrelin let me know if you want to submit PR for this.

**Comment by Ombrelin** [comment]
Indeed that would be very helpful !

**Comment by Ombrelin** [comment]
> @Ombrelin it's working now :D
> Slight issue (not for me), your app uses 300mb+ of ram, I can see a problem with preople with less then 16gb of ram

Glad to hear !
For the memory usage, I'll look into it

**Comment by djguu** [comment]
Thank you guys @Ombrelin and @mclarence ❤️
I would like to help with this, but I don't have enough knowledge of Java, specifically never used a Java framework like spring boot

**Comment by Ombrelin** [comment]
No problem ! Any input is valuable 😉
But don't be afraid of the Spring Boot usage in this project, it's only leveraged for Dependency Injection nothing too fancy

**Comment by djguu** [comment]
@Ombrelin unfortunately, I can't get it to work now, it doesn't capture that I am watching something

**Comment by Ombrelin** [comment]
It says no session detected whereas you have a session playing on PLEX ?

**Comment by djguu** [comment]
yup..
![image](https://user-images.githubusercontent.com/19406042/106656058-1496f480-6592-11eb-9bc5-c52adb4cb5e6.png)

**Comment by mclarence** [comment]
You have multiple plex servers right?

Are you playing the media (shown in the screenshot) from another server or your main one?

Does it work on your other server?

**Comment by djguu** [comment]
tried both, neither one of them showed up in here

**Comment by djguu** [comment]
Before it worked fine, on 1.3

**Comment by Ombrelin** [comment]
It's very weird, since the server fetching process did not change between 1.3 et 1.4, @mclarence correct me if I'm mistaken

**Comment by mclarence** [comment]
No, it has not changed.

I will probably post a new jar file of the app with increased verbosity in the logging so that we can pin point the cause of the issue.

**Comment by mclarence** [comment]
[plex-rich-presence-1.4.jar.zip](https://github.com/Ombrelin/plex-rich-presence/files/5922030/plex-rich-presence-1.4.jar.zip)
 
Please run the above jar file using `java -jar <path to jar file>`. Reproduce the issue and post the log to this issue.

**Comment by mclarence** [comment]
[plex-rich-presence-1.4.zip](https://github.com/Ombrelin/plex-rich-presence/files/5924046/plex-rich-presence-1.4.zip)

Please run the above file with `java -jar plex-rich-presence.jar -Dlogging.level.root=DEBUG` and post the output from the command line here.

**Comment by djguu** [comment]
Sorry I haven't found the time to do it yet, Noticed there was a similar issue with the other user, so I guess it doesn't have a fix?

**Comment by mclarence** [comment]
Yes, if you are not the owner of the plex server then the program will not detect the streams.

However, just post the debug log on pastebin and i'll have a look at it.

**Comment by mclarence** [comment]
> > @Ombrelin it's working now :D
> > Slight issue (not for me), your app uses 300mb+ of ram, I can see a problem with preople with less then 16gb of ram
> 
> Glad to hear the new login fixed your problem !
> For the memory usage, I'll look into it

In regards to memory usage, i've tested it with AdoptOpenJDK 11 + OpenJ9 JVM and it seems to have reduced the usage by quite a bit. From around 250-300 MB to around 160-180MB.

**Comment by Ombrelin** [comment]
That's good news ! I guess you added JavaFX as a Maven dependancy then ?

**Comment by mclarence** [comment]
Yes.

jmetro had to be bumped up to 11.6.14 also.

**Comment by Ombrelin** [comment]
Okay ! My only fear with bumping the Java version is that people most commonly have JRE 8 installed by default.

**Comment by mclarence** [comment]
Maybe you can bundle it with the windows installer.

**Comment by Ombrelin** [comment]
Yes I think there is definitely an option in launch4j to do that

**Comment by Ombrelin** [comment]
Should be fixed in 2.0 version !
